lib/base/process.cpp

```diff
@@ -174,16 +206,65 @@ static Value ProcessSpawnImpl(struct msghdr *msgh, const Dictionary::Ptr& reques
 		sigemptyset(&mask);
 		sigprocmask(SIG_SETMASK, &mask, nullptr);
 
-		if (icinga2_execvpe(argv[0], argv, envp) < 0) {
-			char errmsg[512];
-			strcpy(errmsg, "execvpe(");
-			strncat(errmsg, argv[0], sizeof(errmsg) - strlen(errmsg) - 1);
-			strncat(errmsg, ") failed", sizeof(errmsg) - strlen(errmsg) - 1);
-			errmsg[sizeof(errmsg) - 1] = '\0';
-			perror(errmsg);
-		}
+		for (;;) {
+			if (icinga2_execvpe(argv[0], argv, envp) < 0) {
```

Comment:
All these nested for loops look kind of scary. Wouldn't it be better to truncate the arguments while they are longer than `ARG_MAX`, without calling `icinga2_execvpe` over and over again? Nothing should change in functionality, I would just find it better to read this way!

Reply:
Additionally, all this happens after `fork()`, so you have to be extremely careful regarding what operations are safe to perform. If this would simply happen before, the code should be much simpler.